icechunk/src/storage/mod.rs

```diff
@@ -192,6 +186,30 @@ impl Reader {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum FetchConfigResult {
```

Comment:
I'm also confused. Don't all of these have a generic "network failed" error possibility (for example)? How does that get recorded?